multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ServiceStep.java

```diff
@@ -57,7 +57,12 @@ protected abstract OperationExecutionState executeOperation(ProcessContext conte
 
     protected void processServiceActionFailure(ProcessContext context, CloudServiceInstanceExtended serviceInstance,
                                                CloudOperationException e) {
-        if (!serviceInstance.isOptional()) {
+        if (e.getStatusCode().equals(HttpStatus.CONFLICT)) {
+            getStepLogger().warn(MessageFormat.format(Messages.CREATION_OF_SERVICE_BINDING_OR_KEY_IS_IN_PROGRESS, e.getDescription()));
+            context.setVariable(Variables.IS_SERVICE_BINDING_KEY_CREATION_IN_PROGRESS, true);
```

Comment:
if the variable is set to true, shouldn't the step fail, as it is already has been waited once?